mesonbuild/interpreter/primitives/string.py

```diff
@@ -89,12 +90,10 @@ def endswith_method(self, args: T.Tuple[str], kwargs: TYPE_kwargs) -> bool:
     @noArgsFlattening
     @noKwargs
     @typed_pos_args('str.format', varargs=object)
-    def format_method(self, args: T.Tuple[T.List[object]], kwargs: TYPE_kwargs) -> str:
+    def format_method(self, args: T.Tuple[T.List[TYPE_var]], kwargs: TYPE_kwargs) -> str:
         arg_strings: T.List[str] = []
         for arg in args[0]:
-            if isinstance(arg, bool): # Python boolean is upper case.
-                arg = str(arg).lower()
-            arg_strings.append(str(arg))
+            arg_strings.append(stringifyUserArguments(arg))
```

Comment:
This needs to catch InvalidArguments and fallback to a deprecation warning plus `str()`, because currently this constitutes a breaking change and users will rightly complain that meson is unstable and unreliable and breaks previously working builds that didn't even raise a warning.